### ISSUE TITLE
Using new rustup script

### ISSUE THREAD

**Issue by Razican** [Ogeon/rust-wiringpi#5]
Since we now have rustup (with targets for 1.8.0+) we can now use a simple `cargo build --target=arm-unknown-linux-gnueabihf`.I'm in fact having some problems to update from a previous wiringpi-rs version (1.2 or so): https://travis-ci.org/OpenStratos/OpenStratos-rs/jobs/123140101#L268

I think that now that we have cross-compilation in rustup itself, all those build scripts are not needed (even though we will still need qemu to run tests). So, could the build script be adapted so that we don't have those errors simply by using rustup?

**Comment by Ogeon** [comment]
A simplified build process is worth investigating, but I'm quite sure the `CC` and `AR` variables are still necessary for compiling the C library. It will otherwise use the default values for those, and compile for x86, last I .

I did, in fact, change [the test script for the library](https://github.com/Ogeon/rust-wiringpi/blob/master/.travis.yml) to use multirust. I'm sure it's not as optimal as it could be, but it did work. It should also be possible to move on to testing on stable Rust, now when 1.8 is out.

**Comment by Razican** [comment]
If I have some time to spare next week I can try to do it myself. Seems that the new multirust has big potential :) http://www.rustup.rs/

**Comment by Ogeon** [comment]
That would be great! :+1: I may also get some time to look deeper into this, and also rewrite parts of the [cross compilation guide](https://github.com/Ogeon/rust-on-raspberry-pi), since we don't have to compile Rust ourselves anymore.

**Comment by Razican** [comment]
I'm trying to develop the new build script, and I found this issue: https://travis-ci.org/Razican/rust-wiringpi/jobs/132107556
Seems it fails to include a file named `bits/predefs.h`. Do you know what could it be?

**Comment by Ogeon** [comment]
> I'm trying to develop the new build script

Ah, nice!

> Seems it fails to include a file named `bits/predefs.h`. Do you know what could it be?

It looks like I have had problems with it before, judging by my browsing history, but I can't remember how I solved it. In any case, it seems like it's a part of `libc` and that a missing 32-bit version of it may be an issue. It hasn't been an issue with the current script and tests, so it may be packaged with the Raspberry Pi toolkit. Could it be that your new script doesn't pick those up? I have looked into your fork and you haven't added any replacements for the changed and added environment variables, as far as I can see, so that's likely a contributing factor.
